packages/zk-transfer-web/src/components/Token.tsx

```diff
@@ -1,33 +1,55 @@
 import React, { useState, useCallback, useEffect, useRef } from "react";
 import { note } from "aztec.js"; 
-import { getSum } from "../util";
 import { NoteList, NoteValue } from "../libs/types";
 import arrowLeft from "../assets/arrow-left.svg";
 
 type TokenPropType = {
-  name: string;
+  tradeSymbol: string; // AAA | BBB | ABC
   firstValue?: number;
   secondValue?: number;
   children: any;
   canEdit?: boolean;
-  notes?: NoteList;
+  notes: NoteList;
   onSend: Function;
   allowEdit?: Function;
 };
 
-const fetchValue = (notes: NoteList): Promise<NoteValue[]> => {
-  const getNoteValue = function (e: any) {
-    return note.fromViewKey(e.get("viewingKey"))
-  };
-  const pendingValues: NoteValue[] = notes.map(getNoteValue);
-  return Promise.all(Array.from(pendingValues));
+const useNotes = (notes: NoteList): [number, (Error | null)] => {
+  const [balance, setBalance] = useState(0);
+  const [error, setError] = useState(null);
+
+  const fetchValue = useCallback((notes: NoteList): Promise<NoteValue[]> => {
+    const getNoteValue = function (e: any) {
+      return note.fromViewKey(e.get("viewingKey"))
+    };
+    const pendingValues: any = notes.map(getNoteValue);
+    //@ts-ignore
+    return Promise.all(Array.from(pendingValues));
+  }, []);
+  
+  const fetchNotes = async (notes: NoteList) => {
+      //  console.log('Fetching balances for ', props.tradeSymbol);
+      try {
+        const values = await fetchValue(notes)
+         const amount = values.reduce((s, v) => s + parseInt(v.k), 0)
+         setBalance(amount);
+      } catch (err) {
+        setError(err);
+      }
+  }
+
+  useEffect(() => {
+    fetchNotes(notes);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [notes]);
+
+  return [ balance, error ];
```

Comment:
I prefer `{ balance, error }` here,  so that when you destructure it below as
`{ balance}`, the order doesn't matter.
What do you think?

Reply:
I thought of too. Good one